### PR TITLE
AEA-1533: adding a verify-signature endpoint to the API

### PR DESCRIPTION
**Pull request by callumtipper-IW**

### CODE REVIEW

File: tests/e2e/pact/specs/sandbox/verify-signature.spec.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
+import * as jestpact from "jest-pact"
+import * as uuid from "uuid"
+import {basePath, pactOptions} from "../../resources/common"
+import supertest from "supertest"
+import {fetcher, fhir} from "@models"
+import {InteractionObject} from "@pact-foundation/pact"
+import * as LosslessJson from "lossless-json"
+
+jestpact.pactWith(
+  pactOptions("sandbox", "validate"),
```
Comment (booshibooshi) [inline tests/e2e/pact/specs/sandbox/verify-signature.spec.ts:10]: this needs to match verify.ts ("sandbox", "verify-signature")

Comment (dan-jones-iw) [reply tests/e2e/pact/specs/sandbox/verify-signature.spec.ts:10]: oops ty

File: coordinator/src/services/signature-verification.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+export function verifySignatureHasCorrectFormat(parentPrescription: hl7V3.ParentPrescription): boolean {
+  const signatureRoot = extractSignatureRootFromParentPrescription(parentPrescription)
+  const signature = signatureRoot?.Signature
+  const signedInfo = signature?.SignedInfo
+  const signatureValue = signature?.SignatureValue?._text
+  const x509Certificate = signature?.KeyInfo?.X509Data?.X509Certificate?._text
+  return isTruthy(signedInfo) && isTruthy(signatureValue) && isTruthy(x509Certificate)
+}
+
+export function verifySignatureMatchesPrescription(parentPrescription: hl7V3.ParentPrescription): boolean {
```
Comment (dan-jones-iw) [inline coordinator/src/services/signature-verification.ts:18]: feel like this name is missing something, what you think about `verifySignatureDigestMatchesPrescription`?

**Review by dan-jones-iw — COMMENTED** [review]
theres been no `package.json` changes, should we have `package-lock.json` changes?

File: specification/electronic-prescription-service-api.yaml [hunk 3]
```diff
@@ -450,6 +453,64 @@ paths:
...
+  /FHIR/R4/$verify-signature:
+    post:
+      operationId: verify-signature
+      summary: To verify a prescription's signature
+      description: |
+        ## Overview
+        Use this endpoint to verify a prescription's signature to ensure that the signature is valid, and that it matches the prescription it is attached to.
+        To do this, send a [verify-signature message](TODO) to this endpoint.
+
+        The endpoint will return a set of (up to 25) veify responses associated with the bundle of prescriptions sent to the endpoint for verification.
```
[the remark this answers is not in the document]
Comment (callumtipper-IW) [reply specification/electronic-prescription-service-api.yaml:467]: done

File: coordinator/src/services/signature-verification.ts (hunk 2 above)
Comment (callumtipper-IW) [reply coordinator/src/services/signature-verification.ts:18]: done

File: specification/electronic-prescription-service-api.yaml [hunk 4]
```diff
@@ -188,6 +188,9 @@ info:
     #### To download a prescription:
     POST a [Task-release-message](https://simplifier.net/guide/DigitalMedicines/release-duplicate-2) to the `/FHIR/R4/Task/$release` endpoint.
 
+    #### To verify a prescription's signature:
+    POST a [verify-signature message](TODO) to the `/FHIR/R4/Task/$verify-signature` endpoint.
```
[the remark this answers is not in the document]
Comment (callumtipper-IW) [reply specification/electronic-prescription-service-api.yaml:192]: done

File: specification/electronic-prescription-service-api.yaml [hunk 5]
```diff
@@ -450,6 +453,64 @@ paths:
...
+                  value:
+                    $ref: components/examples/release/not-found-error.json 
+  /FHIR/R4/$verify-signature:
+    post:
+      operationId: verify-signature
+      summary: To verify a prescription's signature
+      description: |
+        ## Overview
+        Use this endpoint to verify a prescription's signature to ensure that the signature is valid, and that it matches the prescription it is attached to.
+        To do this, send a [verify-signature message](TODO) to this endpoint.
```
[the remark this answers is not in the document]
Comment (callumtipper-IW) [reply specification/electronic-prescription-service-api.yaml:465]: done

File: tests/e2e/pact/specs/live/verify-signature.spec.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+            },
+            method: "POST",
+            path: apiPath,
+            body: JSON.parse(messageStr)
+          },
+          willRespondWith: {
+            headers: {
+              "Content-Type": "application/json"
+            },
+            //TODO: add body check
```
[the remark this answers is not in the document]
Comment (callumtipper-IW) [reply tests/e2e/pact/specs/live/verify-signature.spec.ts]: done

**Review by dan-jones-iw — APPROVED** [review]

**Pull request merged**